### PR TITLE
feat(manager/asdf): support `jq`

### PR DESCRIPTION
**Pull request by MPV**
## Changes

Add support for this ASDF plugin:
- https://github.com/lsanwick/asdf-jq

For upgrading versions of `jq`:
- https://github.com/jqlang/jq

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/asdf/extract.spec.ts [hunk 1]
```diff
@@ -303,6 +304,13 @@ dummy 1.2.3
             packageName: 'java-jdk',
             depName: 'java',
           },
+          {
+            currentValue: '1.7',
```
Comment (viceice) [inline lib/modules/manager/asdf/extract.spec.ts:308]: this probably needs a different versioning. did you tested it on a real repo?

Comment (MPV) [reply lib/modules/manager/asdf/extract.spec.ts:308]: Ah. No, not yet (as per the checkboxes in the PR description).

Anything in particular you foresee?

Whenever I find the time I'll learn how to try it on a local repo, and verify.

Comment (rarkins) [reply lib/modules/manager/asdf/extract.spec.ts:308]: The repo includes tags like `jq-1.7rc2`. `semver` won't work for short versions. `semver-coerced` might work, but I don't know if it recognizes those rc ones properly or if it coerces them into a stable version (which would be a problem). `loose` wouldn't recognize or sort properly

Comment (MPV) [reply lib/modules/manager/asdf/extract.spec.ts:308]: How does one (unit) test this with Renovate, that it'll discover versions correctly? Any ideas on how to adjust this PR to solve this?

Comment (viceice) [reply lib/modules/manager/asdf/extract.spec.ts:308]: run on a test repo

Comment (rarkins) [reply lib/modules/manager/asdf/extract.spec.ts:308]: Each of Renovate's versioning schemes includes its own unit tests, so you need to pick one which suits the data/versions that you'll be receiving

Comment (rarkins) [reply lib/modules/manager/asdf/extract.spec.ts:308]: loose versioning might work

Comment (rarkins) [reply lib/modules/manager/asdf/extract.spec.ts:308]: @MPV do you have time to test this?

**Comment by rarkins** [comment]
Closing due to inactivity - PR is still welcome in the future

**Pull request closed without merging**